pkg/kubelet/dockershim/convert.go

```diff
@@ -141,6 +144,9 @@ func toRuntimeAPISandboxState(state string) runtimeApi.PodSandBoxState {
 
 func toRuntimeAPISandbox(c *dockertypes.Container) (*runtimeApi.PodSandbox, error) {
 	state := toRuntimeAPISandboxState(c.Status)
+	if len(c.Names) == 0 {
+		return nil, fmt.Errorf("unexpected empty sandbox name: %+v", c)
```

Comment:
I haven't seen a running docker container that doesn't have a name, so the likelihood should be low?
Even if we do report the container, since it's missing important metadata (name, and who knows what else) associated with the container, kubelet might not be able to recognize it.